spec/controllers/lambda_callback/address_proof_result_controller_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+describe LambdaCallback::AddressProofResultController do
+  describe '#create' do
+    let(:document_capture_session) { DocumentCaptureSession.new(user: create(:user))}
+
+    context 'with valid API token' do
+      before do
+        request.headers['X-API-AUTH-TOKEN'] = Figaro.env.address_proof_result_lambda_token
+      end
+
+      it 'accepts and stores successful address proofing results' do
+        applicant = { phone: Faker::PhoneNumber.cell_phone }
+        document_capture_session.store_proofing_pii_from_doc(applicant)
+        proofer_result = AddressMock.new.proof(applicant)
+
+        post :create, params: {result_id: document_capture_session.result_id,
+                               address_result: proofer_result.to_h}
+        # post address_proof_result_path(result_id: document_capture_session.result_id),
+        #   params: {result_id: '1', a: 'b'}
+
+        proofing_result = document_capture_session.load_proofing_result
+        expect(proofing_result.pii).to eq nil
+        expect(proofing_result.result).to eq({ exception: "", success: 'true' })
+      end
+
+      it 'accepts and stores unsuccessful address proofing results' do
+        applicant = { phone: '7035555555' }
+        document_capture_session.store_proofing_pii_from_doc(applicant)
+        proofer_result = AddressMock.new.proof(applicant)
+
+        post :create, params: {result_id: document_capture_session.result_id,
+                               address_result: proofer_result.to_h}
+
+        # post address_proof_result_path(result_id: document_capture_session.result_id),
```

Comment:
I tried for a while, but I could not get the path parameter to work in the specs. I don't know if it's a weird rails thing, but it worked fine in dev via curl and errored saying it couldn't find the path in the tests.

Reply:
Controller specs shouldn't use paths like this, they should use `post :create` etc because the controller is inferred

But I commented out the other line, and this worked in the rails console:

```rb
Rails.application.routes.url_helpers.address_proof_result_url(result_id: 'aaa')
=> "http://localhost:3000/api/address_proof_result/aaa"
```

Reply:
I initially tried using `post :create`, but couldn't figure out how to pass both the `:result_id` as a path parameter and the body

Turns out you can specify all of them combined in params and rails figures out which go where:

Reply:
yup! sorry didn't realize that was the issue